ssz-rs/src/merkleization/mod.rs

```diff
@@ -21,17 +19,39 @@ pub trait Merkleized {
     fn hash_tree_root(&mut self) -> Result<Node, MerkleizationError>;
 }
 
-#[derive(Error, Debug)]
-#[error("the value could not be merkleized: {0}")]
+#[derive(Debug)]
 pub enum MerkleizationError {
-    #[error("failed to serialize value: {0}")]
-    SerializationError(#[from] SerializeError),
-    #[error("cannot merkleize a partial chunk of length {1} (data: {0:?})")]
+    SerializationError(SerializeError),
     PartialChunk(Vec<u8>, usize),
-    #[error("cannot merkleize data that exceeds the declared limit {0}")]
     InputExceedsLimit(usize),
 }
 
+impl From<SerializeError> for MerkleizationError {
+    fn from(e: SerializeError) -> Self {
+        MerkleizationError::SerializationError(e)
+    }
+}
+
+impl Display for MerkleizationError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        match self {
+            MerkleizationError::SerializationError(e) => {
+                write!(f, "failed to serialize value: {}", e)
```

Comment:
if we look at the inner error message, we don't need to add any additional text here
```suggestion
                write!(f, "{e}")
```